x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/relevance/boosts/ProximityBoost.java

```diff
@@ -0,0 +1,225 @@
+
+    public enum CenterType {
+        Numeric,
+        Geo,
+        Date
+    }
+
+    private CenterType centerType = null;
+
+    public ProximityBoost(String center, String function, Float factor) {
+        super(TYPE, "add");
+        this.center = center;
+        this.function = function;
+        this.factor = factor;
+        this.setCenterType();
+    }
+
+    public void setCenterType() {
+        centerType = null;
+        if (isNumber()) {
+            centerType = CenterType.Numeric;
+            return;
+        }
+        if (isDate()) {
+            centerType = CenterType.Date;
+            return;
+        }
+        if (isGeo()) {
+            centerType = CenterType.Geo;
+        }
+    }
+
+    private NumberFormat getNumberFormat() {
+        return NumberFormat.getInstance(Locale.ROOT);
+    }
+
+    public boolean isGeo() {
+        try {
+            String[] latLon = center.split(",", 2);
+            getNumberFormat().parse(latLon[0].trim());
+            getNumberFormat().parse(latLon[1].trim());
+            return true;
+        } catch (Exception e) {
+            // do nothing
+        }
+        try {
+            Geohash.decodeLatitude(center);
+            Geohash.decodeLongitude(center);
+            return true;
+        } catch (Exception e) {
+            // do nothing
+        }
+        try {
+            WellKnownText.fromWKT(geoValidator, true, center);
+            return true;
+        } catch (Exception e) {
+            // do nothing
+        }
+        return false;
+    }
+
+    public boolean isNumber() {
+        try {
+            Double.valueOf(center);
+            return true;
+        } catch (Exception x) {
+            // do nothing
+        }
+        try {
+            Integer.valueOf(center);
+            return true;
+        } catch (Exception x) {
+            // do nothing
+        }
+        return false;
+    }
+
+    public boolean isDate() {
+        try {
+            dateParser.parse(center, () -> 0);
+            return true;
+        } catch (ElasticsearchParseException e) {
+            return false;
+        }
+    }
+
+    public String getCenter() {
+        return center;
+    }
+
+    public Number getNumericCenter() throws ParseException {
+        return getNumberFormat().parse(center);
+    }
+
+    public String getDateCenter() {
+        if (Objects.equals(center, "now")) {
+            return dateParser.parse(center, () -> 0).toString();
+        }
+        return center;
+    }
+
+    public String getFunction() {
+        return function;
+    }
+
+    public Float getFactor() {
+        return factor;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        ProximityBoost that = (ProximityBoost) o;
+        return (this.function.equals(that.getFunction()) && this.center.equals(that.getCenter()) && this.factor.equals(that.getFactor()));
```

Comment:
`hashCode()` includes the hash of `type`, should that be included here as well for parity?